include/alpaka/vec/Vec.hpp

```diff
@@ -169,6 +169,8 @@ namespace alpaka
             typename = typename std::enable_if<
                 // There have to be dim arguments.
                 (sizeof...(TArgs)+1 == TDim::value)
+                &&
+                (std::is_same<TSize, TArg0>::value)
```

Comment:
I think we can not replace the line 176 with the new line, because line 176 is in a disjunction with line 178 and line 178 would return true for non alpaka vectors. But we might remove line 176.